spec/helpers/application_helper_spec.rb

```diff
@@ -78,4 +78,31 @@
       it { is_expected.to have_css "body.homepage" }
     end
   end
+
+  describe "#tta_service_link" do
+    let(:tta_url) { "https://tta-service/" }
+    before { allow(ENV).to receive(:[]).and_call_original }
+    before { allow(ENV).to receive(:[]).with("TTA_SERVICE_URL") { tta_url } }
+    subject { tta_service_link(class: "test") { "Testing" } }
+
+    context "with link set" do
+      it { is_expected.to have_css "a[href=\"https://tta-service/\"]" }
+      it { is_expected.to have_css "a.test" }
+      it { is_expected.to have_css "a", text: "Testing" }
+    end
+
+    context "without link set" do
+      let(:tta_url) { nil }
+      it { is_expected.to be_nil }
+    end
+
+    context "with username and password set" do
+      before do
+        allow(ENV).to receive(:[]).with("HTTPAUTH_USERNAME").and_return "user"
+        allow(ENV).to receive(:[]).with("HTTPAUTH_PASSWORD").and_return "pass"
+      end
+
+      it { is_expected.to have_css "a[href=\"https://user:pass@tta-service/\"]" }
```

Comment:
Is this to avoid the user being prompted for the basic auth creds when navigating to the TTA service? Presumably the browser history will then contain these for others to see, unless it's filtered out by modern browsers? is that a problem?